src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1699,6 +1705,23 @@ class DocBaseViewer extends BaseViewer {
             this.cache.set(DOCUMENT_FTUX_CURSOR_SEEN_KEY, true, true);
         }
     }
+
+    /**
+     * Get a thumbnail image element
+     *
+     * @param {number} pageNumber - the page number
+     * @return {Promise} - promise resolves with the image HTMLElement or null if generation is in progress
+     */
+    getThumbnail(pageNumber) {
+        if (!this.thumbnailsSidebar) {
+            this.thumbnailsSidebar = new ThumbnailsSidebar(this.thumbnailsSidebarEl, this.pdfViewer);
+        }
+        return this.thumbnailsSidebar.createThumbnailImage(pageNumber - 1, {
```

Comment:
Should we instead make `createThumbnailImage` a static method on the `ThumbnailsSidebar` class?

Reply:
I think it’s better to not make it static. We need to show page thumbnails on the CI modal, and to accomplish this, we need to use PDFJS to fetch those. We still need to get proper thumbnail representation (WiP). I used the ThumbnailSidebar to avoid repeating the code for fetching a thumb, so my approach is to not change this class a lot because it is being used as a workaround at this moment. (Changing the createThumbnailImage to a static method would need other methods to be updated as well)